src/js/components/List.jsx

```diff
@@ -0,0 +1,197 @@
+import React, { PropTypes } from 'react';
+import KeyCodes from '../KeyCodes';
+
+type Props = {
+    className?: string,
+    children?: ReactChildren,
+    style?: Object,
+    data?: Array<Object>,
+    labelFunction?: Function,
+    defaultStyles: bool,
+    defaultContainerClass: bool,
+    defaultItemClass: bool,
+    defaultSelection?: Object,
+    onItemSelect?: Function,
+}
+
+type State = {
+    selectedItem?: Object,
+};
+
+// TODO: fix a few flow errors
+/*
+type RendererProps = {
+    listIndex: number,
+    listItem: string | Object,
+    labelFunction: Function,
+}
+*/
+
+
+export class List extends React.Component {
+
+    props: Props;
+    state: State;
+
+    static defaultProps: Props = {
+        style: {},
+        defaultStyles: true,
+        defaultContainerClass: true,
+        defaultItemClass: true,
+    };
+
+    constructor(props:Props) {
+        super(props);
+
+        this.state = {
+            selectedItem: null,
+        };
+    }
+
+    componentWillReceiveProps(nextProps:Props) {
+        if (this.props.data !== nextProps.data) {
+            this.setState({
+                selectedItem: null,
+            });
+        }
+
+        this._defaultSelection(nextProps);
+    }
+
+    componentDidMount() {
+        this._defaultSelection(this.props);
+    }
+
+    get selectedIndex():number {
+        return this.props.data ? this.props.data.indexOf(this.state.selectedItem) : -1;
+    }
+
+    get selectedItem():Object {
+        return this.state.selectedItem;
+    }
+
+    _defaultSelection(nextProps:Props) {
+        if (!this.state.selectedItem && nextProps.defaultSelection) {
+            this.setState({
+                selectedItem: nextProps.defaultSelection,
+            });
+        }
+    }
+
+    _onClickListItem(event:Event, index:number, item:Object) {
+        event.preventDefault();
+
+        this._selectItem(index, item);
+    }
+
+    _onKeyPressListItem(event:React$Event, index:number, item:Object) {
+        event.preventDefault();
+
+        if (event.nativeEvent.keyCode === KeyCodes.SPACEBAR) {
```

Comment:
So, I got a bit lazy on my list implementation before. I think the keyboard interaction with this should be a bit different, basically like that of radio buttons, tab to the list, use arrow keys to navigate the items within it. Also, not sure the default behavior should be to select an item in the list but rather just call a callback and let the caller decide what happens. For example, I already had to change the behavior of my list to click/enter/etc. select and dismiss the containing dialog.